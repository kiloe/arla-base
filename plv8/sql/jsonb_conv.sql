CREATE FUNCTION convb(o jsonb) RETURNS jsonb AS $$
if (o instanceof Array) {
	o[1] = 10;
} else if (typeof(o) == 'object') {
	o.i = 10;
}
return o;
$$ LANGUAGE plv8;

SELECT convb('{"i": 3, "b": 20}');
SELECT convb('[1, 2, 3]');

CREATE FUNCTION get_keyb(key text, json_raw jsonb) RETURNS jsonb
LANGUAGE plv8 IMMUTABLE STRICT
AS $$
  var val = json_raw[key];
  var ret = {};
  ret[key] = val;
  return ret;
$$;

CREATE TABLE jsonbonly (
    data jsonb
);

COPY jsonbonly (data) FROM stdin;
{"ok": true}
\.

-- Call twice to test the function cache.
SELECT get_keyb('ok', data) FROM jsonbonly;
SELECT get_keyb('ok', data) FROM jsonbonly;
