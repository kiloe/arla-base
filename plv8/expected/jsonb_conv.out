CREATE FUNCTION convb(o jsonb) RETURNS jsonb AS $$
if (o instanceof Array) {
	o[1] = 10;
} else if (typeof(o) == 'object') {
	o.i = 10;
}
return o;
$$ LANGUAGE plv8;
SELECT convb('{"i": 3, "b": 20}');
       convb        
--------------------
 {"b": 20, "i": 10}
(1 row)

SELECT convb('[1, 2, 3]');
   convb    
------------
 [1, 10, 3]
(1 row)

CREATE FUNCTION get_keyb(key text, json_raw jsonb) RETURNS jsonb
LANGUAGE plv8 IMMUTABLE STRICT
AS $$
  var val = json_raw[key];
  var ret = {};
  ret[key] = val;
  return ret;
$$;
CREATE TABLE jsonbonly (
    data jsonb
);
COPY jsonbonly (data) FROM stdin;
-- Call twice to test the function cache.
SELECT get_keyb('ok', data) FROM jsonbonly;
   get_keyb   
--------------
 {"ok": true}
(1 row)

SELECT get_keyb('ok', data) FROM jsonbonly;
   get_keyb   
--------------
 {"ok": true}
(1 row)

